src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -175,7 +174,7 @@ DistributedTableSize(Oid relationId, char *sizeQuery)
 
 	ErrorIfNotSuitableToGetSize(relationId);
 
-	pgDistNode = heap_open(DistNodeRelationId(), AccessShareLock);
+	LockRelationOid(DistNodeRelationId(), AccessShareLock);
 
 	workerNodeList = ActivePrimaryNodeList();
```

Comment:
It might make sense to put the lock into `ActivePrimaryNodeList`. I can't really come up with a reason not do so.